src/codemirror/utils/extensions-utils.ts

```diff
@@ -0,0 +1,26 @@
+import { EditorState } from "@uiw/react-codemirror";
+
+interface EditorStateInfo {
+  cursorPos: number;
+  fullLineText: string;
+  lineTextUpToCursor: string;
+  lineTextAfterCursor: string;
+}
+
+export function parseTagFromLineText(lineText: string): string {
+  return lineText.match(/<([a-zA-Z0-9]+)/)?.[1];
+}
+
+export function extractEditorStateInfo(state: EditorState): EditorStateInfo {
```

Comment:
```suggestion
export function getEditorStateInfo(state: EditorState): EditorStateInfo {
```
it's just a getter no?